packages/utilities/src/DaemonUtils.ts

```diff
@@ -0,0 +1,151 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+
+/**
+ * Protocol for daemon mode is to send headers similar to those used in HTTP.  All headers
+ * must appear on the same line without a newline.  The version 1 of this protocol requires
+ * that all 7 original headers be present.  The first header must always be x-zowe-daemon-headers
+ * and the last must always be x-zowe-daemon-end.
+ *
+ * Integers are the only content passed.  For boolean header types, 0 = false, 1 = true.
+ */
+
+import { IDaemonHeaderOptions } from "./doc/IDaemonHeaderOptions";
+
+/**
+ * Class to construct headers for daemon mode protocol
+ * @export
+ * @class DaemonUtils
+ */
+export class DaemonUtils {
+
+
+    /**
+     * Header to indicate the count of total headers being sent including this header itself
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_DAEMON_HEADERS = "x-zowe-daemon-headers:";
+
+    /**
+     * Current version of the headers.  Version 1 = the first 7 headers.
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_DAEMON_VERSION = "x-zowe-daemon-version:";
+
+    /**
+     * Alternate process exit code.  Default is 1.
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_DAEMON_EXIT = "x-zowe-daemon-exit:";
+
+    /**
+     * Prompting is requested from the daemon to the client if 1.  Default is 0.
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_DAEMON_PROMPT = "x-zowe-daemon-prompt:";
+
+    /**
+     * Content is for stdout if 1.  Default is 1.
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_DAEMON_STDOUT = "x-zowe-daemon-stdout:";
+
+    /**
+     * Content is for stderr if 1. Default is 0.
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_DAEMON_STDERR = "x-zowe-daemon-stderr:";
+
+    /**
+     * Ending header.  Value does not matter.
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_DAEMON_END = "x-zowe-daemon-end:";
+
+    /**
+     * Version 1 of the headers means there are 7 total.
+     * @static
+     * @memberof DaemonUtils
+     */
+    public static readonly X_ZOWE_HEADERS_V1 = 1;
```

Comment:
This is a minor item about variable names:
    X_ZOWE_HEADERS_V1 = 1
    X_ZOWE_HEADERS_V1_LEN = 7

The name and the value both appear to lock the variable name into version 1. If we ever produced version 2, which change below would feel inclined to make?
    X_ZOWE_HEADERS_V1 = 2
or also change the variable name?
    X_ZOWE_HEADERS_V2 = 2

Reply:
I'll rename the two you pointed out.   `X_ZOWE_HEADERS_V1` and `X_ZOWE_HEADERS_V1_LEN` are initial values, not headers.  

X_ZOWE_DAEMON_VERSION is supposed to hold the first version.